src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -533,6 +553,8 @@ void Terminal::_InitializeColorTable()
 // - isVisible: whether the cursor should be visible
 void Terminal::SetCursorVisible(const bool isVisible) noexcept
 {
+    auto lock = LockForReading();

```

Comment:
Given that you're setting something about the buffer, this would be a `LockForWriting()`, not reading.